StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1214StaticReadonlyElementsMustAppearBeforeStaticNonReadonlyElements.cs

```diff
@@ -43,7 +46,61 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(this.AnalyzeClass, SyntaxKind.ClassDeclaration);
+            context.RegisterSyntaxNodeAction(this.AnalyzeStruct, SyntaxKind.StructDeclaration);
+        }
+
+        private void AnalyzeStruct(SyntaxNodeAnalysisContext context)
```

Comment:
:exclamation: Rather than separate methods `AnalyzeStruct`, `AnalyzeClass`, and `AnalyzeType`, can you create one method `HandleTypeDeclaration` which starts with this:

```csharp
var typeDeclaration  = (TypeDeclarationSyntax)context.Node;
```

You can then register this method as the handler for both `ClassDeclaration` and `StructDeclaration`.